File: src/main/java/com/starshootercity/abilities/Invisibility.java
package com.starshootercity.abilities;

import com.starshootercity.OriginSwapper;
import net.kyori.adventure.key.Key;
import org.bukkit.entity.Player;
import org.jetbrains.annotations.NotNull;

import java.util.List;

public class Invisibility implements DependantAbility, VisibleAbility, VisibilityChangingAbility {
    @Override
    public @NotNull Key getKey() {
        return Key.key("origins:invisibility");
    }

    @Override
    public @NotNull Key getDependencyKey() {
        return Key.key("origins:phantomize");
    }

    @Override
    public @NotNull List<OriginSwapper.LineData.LineComponent> getDescription() {
        return OriginSwapper.LineData.makeLineFor("While phantomized, you are invisible.", OriginSwapper.LineData.LineComponent.LineType.DESCRIPTION);
    }

    @Override
    public @NotNull List<OriginSwapper.LineData.LineComponent> getTitle() {
        return OriginSwapper.LineData.makeLineFor("Invisibility", OriginSwapper.LineData.LineComponent.LineType.TITLE);
    }

    @Override
    public boolean isInvisible(Player player) {
        return getDependency().isEnabled(player);
    }
}

File: core/src/main/java/com/starshootercity/abilities/Invisibility.kt
package com.starshootercity.abilities

import com.starshootercity.OriginSwapper.LineData.Companion.makeLineFor
import com.starshootercity.OriginSwapper.LineData.LineComponent
import net.kyori.adventure.key.Key
import org.bukkit.entity.Player

class Invisibility : DependantAbility, VisibleAbility, VisibilityChangingAbility {

    override fun getKey(): Key {
        return Key.key("origins:invisibility")
    }

    override val dependencyKey: Key = Key.key("origins:phantomize")

    override val description: MutableList<LineComponent?> = makeLineFor(
        "While phantomized, you are invisible.",
        LineComponent.LineType.DESCRIPTION
    )

    override val title: MutableList<LineComponent?> = makeLineFor(
        "Invisibility",
        LineComponent.LineType.TITLE
    )

    override fun isInvisible(player: Player): Boolean {
        return dependency.isEnabled(player)
    }
}
